packages/frontend/src/hooks/contracts/useTokenBalance.ts

```diff
@@ -27,16 +27,15 @@ export const useTokenBalance = (token: string, refetchIntervalSec = 30, decimals
   }, [web3, token])
 
   const balanceQuery = useQuery(
-    tokenBalanceQueryKeys.userTokenBalance(),
+    tokenBalanceQueryKeys.userTokenBalance(token),
     () => updateBalance(token, connected, contract, address, decimals),
     {
-      enabled: Boolean(token) && Boolean(connected) && Boolean(contract),
+      enabled: Boolean(token) && Boolean(connected) && Boolean(contract) && !wait,
       refetchInterval: refetchIntervalSec * 15000,
-      staleTime: 15000,
```

Comment:
Instead of hard coded stale time, we'll wait until the query is ready to start fetching